packages/server/src/api/controllers/row.js

```diff
@@ -229,6 +229,36 @@ exports.fetchView = async function(ctx) {
   }
 }
 
+exports.search = async function(ctx) {
+  const appId = ctx.user.appId
+
+  const db = new CouchDB(appId)
+
+  const {
+    query,
+    pagination: { pageSize = 10, page },
+  } = ctx.request.body
+
+  query.tableId = ctx.params.tableId
+
+  const response = await db.find({
+    selector: query,
+    limit: pageSize,
+    skip: pageSize * page,
+  })
+
+  const rows = response.docs
+
+  // delete passwords from users
+  if (query.tableId === ViewNames.USERS) {
+    for (let row of rows) {
+      delete row.password
+    }
+  }
+
+  ctx.body = await linkRows.attachLinkInfo(appId, rows)
```

Comment:
Minor note - in master instead of doing the `attachLinkInfo` all calls which return rows should instead call `enrichRows` from utilities, found [here](https://github.com/Budibase/budibase/blob/4f1546d0573191136d3189e2bccf151a39ba87a2/packages/server/src/utilities/index.js#L226). This does require getting the table as well, as it needs to understand the schema to enrich, this is purely for attachment URLs.

Reply:
Makes sense - will update 👍